src/libosd/include/osd/cl_dem_uart.h

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OSD_CL_DEM_UART_H
+#define OSD_CL_DEM_UART_H
+
+#include <stdlib.h>
+
+#include <osd/osd.h>
+#include <osd/packet.h>
+#include <osd/hostmod.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct osd_dem_uart_ctx {
+	struct osd_hostmod_ctx *hostmod_ctx;
+	unsigned int local_diaddr;
+	unsigned int remote_diaddr;
+};
+
+struct osd_dem_uart_event {
+	char character;
+};
+// Maybe later we could use a struct like this?
+/*struct osd_dem_uart_event {
+	uint8_t num_chars;
+	uint8_t  chars[0];
+};*/
+
+typedef void (*osd_cl_dem_uart_handler_fn)(
+	void * /*cb_arg*/, struct osd_dem_uart_event * /*event to be handled*/);
+
+struct osd_dem_uart_event_handler {
+	struct osd_dem_uart_ctx * cb_ctx;
+	osd_cl_dem_uart_handler_fn cb_fn;
+	void *cb_arg;
+};
+
+struct osd_dem_uart_desc {
+	unsigned int diaddr;
```

Comment:
we typically use `di_addr` (it's not completely consistent, but this naming makes it at least consistent within this file).